Frontend/src/Login.jsx

```diff
@@ -1,45 +1,140 @@
 import { useState } from "react";
-import logo from "./img/temp.png"
+import axios from "axios"
+import logo from "./img/title.png"
+import login_button from "./img/Login_Button.png"
+import create_account from "./img/Create_Account.png"
+import new_user from "./img/New_User.png"
+import exist_user from "./img/Existing_User.png"
+import PropTypes from "prop-types";
 import "./css/Login.css"
 
-function Login() {
+const API_URL = "http://localhost:3500"
+var loginAcc = login_button
+var userState = new_user
+export default function Login({setLoginTime, apiKey, setApiKey, setUserData}) {
   const [username, setUser] = useState("");
   const [password, setPassword] = useState("");
+  const [logState, setLogState] = useState("login");
+
   const handleSubmit = (e) => {
-      e.preventDefault();
-      //login API/function
-      alert("Username: " + username + " " + "Password: " + password);
+    e.preventDefault();
+    attemptLogin(logState);
+    // alert("Username: " + username + " " + "Password: " + password);
   };
+
+
+  const attemptLogin = async (method) => {
+    setLoginTime(Date.now());
+
+    if (method === "login") {
+      try {
+        const temp = await axios.get(`${API_URL}/loginUser`, {
+          params: {
+            "username": username,
+            "password": password
+          }
+        })
+
+        setApiKey(localStorage.getItem("canvasAPIKey"))
+        setUserData(temp.data.u)  //dont know why I need a u here
+      } catch (e) {
+        console.log("login failed");
+      }
+
+    } else if (logState === "create") {
+      try {
+        const temp = await axios.get(`${API_URL}/registerAccount`, {
+          params: {
+            "username": username,
+            "password": password
+          }
+        })
+        localStorage.setItem("canvasAPIKey", apiKey)
+
+        //LOG IN THE USER AS WELL
+        await setLogState("login");
+        attemptLogin("login"); //logState doesn't update by func call for some reason
+
+      } catch (e) {
+        console.log("account creation failed") //TELL THE USER IF THE USERNAME IS TAKEN
+      }
+    }
+  }
+
+  const toggleLoginState = () => {
+    if (logState === "login") {
+      setLogState("create")
+      loginAcc = create_account
+      userState = exist_user
+    } else {
+      setLogState("login")
+      loginAcc = login_button
+      userState = new_user
+    }
+  }
```

Comment:
can the `loginAcc` and `userState` changes be moved into the view itself (like with a ternary operator)? this would help avoid issues like the one above